### PR TITLE
Force recommended results to the top when sorting from a Category page

### LINKED ISSUE #8084: Force recommended results to the top when sorting from a Category page
This is a spin-off from #8079.

When a user is on a Search Results page which includes `category` in the search criteria, if they change the sort criteria using the form, ensure that recommended add-ons remain at the top of the list by changing the sort criteria sent to the API to include `recommended` as the first criterion.

This is required because we are doing away with Category pages and they are being replaced with Search Results. See #8079 for more info.

### PR DESCRIPTION
**Pull request by bobsilverberg**
Fixes #8084

### CODE REVIEW

File: src/amo/components/SearchFilters/index.js [hunk 1]
```diff
@@ -30,6 +31,7 @@ import Select from 'ui/components/Select';
 import './styles.scss';
 
 const NO_FILTER = '';
+const sortSelectName = 'sort';
```
Comment (willdurand) [inline src/amo/components/SearchFilters/index.js]: Could you use `paramsToFilter.sort` from `src/core/searchUtils.js` instead?

File: tests/unit/amo/components/TestSearchFilters.js [hunk 2]
```diff
@@ -109,6 +112,116 @@ describe(__filename, () => {
...
+      filters: {
+        query: 'Music player',
+        sort,
+      },
+    });
+
+    expect(root.find('.SearchFilters-Sort')).toHaveProp('value', sort);
+  });
+
+  it('selects the second sort criterion in the sort select when there are two criteria', () => {
```
Comment (willdurand) [inline tests/unit/amo/components/TestSearchFilters.js]: Shouldn't we try to select the criterion that is not `recommended` instead or should we have a `multiple` select?

File: src/amo/components/SearchFilters/index.js (hunk 1 above)
Comment (bobsilverberg) [reply src/amo/components/SearchFilters/index.js]: Good idea. I think I'll keep `sortSelectName` but set it to `paramsToFilter.sort`. I'm not sure if you meant to get rid of it entirely and just use `paramsToFilter.sort` in the code, but I think having the name makes it clearer that there's a relationship between the two uses.

File: tests/unit/amo/components/TestSearchFilters.js (hunk 2 above)
Comment (bobsilverberg) [reply tests/unit/amo/components/TestSearchFilters.js]: I like the idea of making the the first criterion that is not `recommended`, so I've implemented that change. I don't think we want to complicate it by allowing the user to select multiple sort criteria (in which case they'd also have to be able to specify which is the first sort and which is the second), but we can open that up for debate and a follow-up could be filed for that.

@jvillalobos Do you think we should enhance the sort form to allow users to specify multiple sort fields (now that we're sometimes forcing the `recommended` sort as the first criterion)?

Comment (jvillalobos) [reply tests/unit/amo/components/TestSearchFilters.js]: No, I don't think that will be necessary. Prioritizing recommended extensions should be invisible to the user.

Comment (bobsilverberg) [reply tests/unit/amo/components/TestSearchFilters.js]: > No, I don't think that will be necessary. Prioritizing recommended extensions should be invisible to the user.

Well, it won't be invisible if they look at the URL, but it won't be evident from the form. I assume that is acceptable?

File: src/amo/components/SearchFilters/index.js (hunk 1 above)
Comment (willdurand) [reply src/amo/components/SearchFilters/index.js]: It's fine to keep `sortSelectName` :)

**Pull request merged**